Tests/test_death_handling.gd

```diff
@@ -116,3 +134,16 @@ func test_enemy_death_to_poison():
 	assert_eq(_enemy_list.size(), 2)
 	_battler._on_enemy_start_turn()
 	assert_eq(_enemy_list.size(), 1)
+
+
+func test_enemy_death_to_expiring_poison():
+	_enemy_health_component._set_health(1.0)
+	var card_poison: CardBase = load("res://Cards/Resource/Card_Poison.tres")
+	card_poison.on_card_play(_player, [_enemy])
+	_enemy.get_status_component().current_status[0].status_turn_duration = 1
+
+	assert_eq(_enemy_list.size(), 2)
+	assert_eq(_enemy.get_status_component().current_status.size(), 1)
+	_battler._on_enemy_start_turn()
+	assert_eq(_enemy.get_status_component().current_status.size(), 0)
```

Comment:
does that even work ? isn't the enemy dead thus the resource is freed and you can't check his status component ?

Reply:
freeing happens deferred via queue_free so it doesn't actually get freed until the next frame - so at the end of the test the enemy is still a valid node and we can check it - at least that's my understanding and the test has been consistently passing but let me know if you think I'm wrong in something

Actually I don't even see a way to check if enemy has actually really been freed in a test (which I wanted to assert for as well) because I would have to await for the next frame in the test - which I tried to do but it then broke things as it seemed to me that the "_before_each_" was executed when the next test was already in progress and it freed the node midrun

Reply:
`before_each` is executed at the start of each test, so if you await in a specific test, it shouldn't just jump to the next test and launch the `before_each`

`_on_enemy_start_turn` calls `_handle_deaths` which calls `_handle_enemy_deaths`, and in there you add everything to a list to remove, erase and queue free. `queue_free` actually frees at the end of the current frame. If that's the only thing you need to wait you can either:
- `await get_tree().process_frame` to wait for the deferred calls
- `is_queued_for_deletion()` doesn't wait for deferred calls but you can check the enemy is indeed waiting to disappear
Option one is probably better as it makes sure everything works instead of assuming the `queue_free` works directly

Reply:
When I try to do it via option one, which is essentially the same as I did before which I described above, it results in tests throwing error "Nonexistent function in base 'Nil'" at some point during test's execution - which the only way I can see happening is if the player was freed in the middle of the test run and the only way we are freeing it is at the end of the previous test 🤔

Option 2 works fine

Reply:
Did you try to put some breakpoints in godot to check when things are being freed ? That might allow you to see why option one doesn't work

Reply:
It seems the issue is PhaseManager is being recreated at a wrong time somehow and switches phase to Player turn which tries to apply player statuses at a time when player doesn't exist
![image](https://github.com/Saplings-Projects/1M_sub/assets/24390212/82951cbd-4a91-4cad-a8aa-f6e1e27cb178)


Also it looks like the issue is happening if I use both await inside of tests AND queue_free in _after_each_. I tried switching the freeing after tests to _free_ and it seems like it's passing without issues

Reply:
well as long as it's passing consistently it's fine

Reply:
Added asserts into tests - I kept both queued_for_deletion before await and checking if node was freed after